includes/Core/Storage/Post_Meta_Setting.php

```diff
@@ -33,15 +33,26 @@ abstract class Post_Meta_Setting {
 	 */
 	protected $post_meta;
 
+	/**
+	 * Arguments for `register_meta` call.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	protected $args = array();
+
 	/**
 	 * Post_Meta_Setting constructor.
 	 *
-	 * @since 1.33.0
+	 * @since 1.33.0 Function introduced.
+	 * @since n.e.x.t Added the $args option.
 	 *
 	 * @param Post_Meta_Interface $post_meta Post_Meta_Interface instance.
+	 * @param array               $args      Options to pass to `register_meta`.
 	 */
-	public function __construct( Post_Meta_Interface $post_meta ) {
+	public function __construct( Post_Meta_Interface $post_meta, $args = array() ) {
```

Comment:
@tofumatt, although this approach works, we should better solve the problem with the `show_in_rest` option using the OOP approach. What we need to do is the following:

1. Remove the new `$args` approach.
1. Add a new protected method `get_show_in_rest` that returns `false` by default.
1. Update the `register_meta` function call to be as the following:
    ```php
    register_meta(
        'post',
        static::META_KEY,
        array(
            'type'              => $this->get_type(),
            'sanitize_callback' => $this->get_sanitize_callback(),
            'single'            => true,
            'show_in_rest'      => $this->get_show_in_rest(),
        )
    );
    ```
1. Add the new method `get_show_in_rest` to the `Post_Idea_Text` class that returns `true`.

Reply:
Right on, that's nicer 🙂